gprofiler/profilers/python.py

```diff
@@ -371,6 +423,11 @@ def snapshot(self) -> ProcessToStackSampleCounters:
         parsed = parse_many_collapsed(collapsed_text)
         if self.add_versions:
             parsed = _add_versions_to_stacks(parsed)
+        for pid in parsed:
+            try:
+                PythonMetadta.update_metadata(Process(pid), self._stop_event)
+            except NoSuchProcess:
```

Comment:
Well technically the `Process(pid)` will raise `NoSuchProcess`, that's what I had in mind :sweat_smile: 

Also, if we raise in `_get_python_version`, then I am catching it in `make_application_metadata`.

However, I do admit that I haven't done a pass on all executed code to ensure no exceptions leak. Looking at it now... we can also fail when reading the map files / exe, and that'd be just `FileNotFoundError`. Not sure what's the correct approach here - a catch-all at the `make_application_metadata` level, or trying to catch each of those exceptions where it might be raised... Wdyt?

Reply:
> Well technically the Process(pid) will raise NoSuchProcess, that's what I had in mind 😅

Racy

I think we should except `NoSuchProcess` and `continue` (like now), and also except-all in here, specifically for `PythonEbpfProfiler` and log, so we won't interfere with the rest of the processes, this isn't a concern for `ProcessProfiler`s.

And to reduce the chance for a race and redundant logs - we should catch the exceptions where they might be raised

Reply:
> Well technically the Process(pid) will raise NoSuchProcess, that's what I had in mind 😅

Racy

I think we should except `NoSuchProcess` and `continue` (like we do now), and also except-all in here, specifically for `PythonEbpfProfiler` and log, so we won't interfere with the rest of the processes, this isn't a concern for `ProcessProfiler`s.

Catching all in `make_application_metadata` will not allow us to `except NoSuchProcess` in `PythonEbpfProfiler` that we need...

And to reduce the chance for a race and redundant logs - we should catch the exceptions where they might be raised

Reply:
I am swallowing all exceptions besides `NoSuchProcess` now.